File: nhom4_QLBA_DLH/lib/models/MedicalRecordDetails.dart
class MedicalRecordDetails {
  final int id;
  final int patientId;
  final int appointmentDetailId;
  final String? treatment;
  final DateTime? createdAt;

  MedicalRecordDetails({
    required this.id,
    required this.patientId,
    required this.appointmentDetailId,
    this.treatment,
    this.createdAt,
  });

  factory MedicalRecordDetails.fromJson(Map<String, dynamic> json) {
    return MedicalRecordDetails(
      id: json['id'],
      patientId: json['patientId'],
      appointmentDetailId: json['appointmentDetailId'],
      treatment: json['treatment'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'patientId': patientId,
      'appointmentDetailId': appointmentDetailId,
      'treatment': treatment,
      'createdAt': createdAt?.toIso8601String(),
    };
  }
}

File: Nhom4_QLBA_API/Nhom4_QLBA_API/Models/MedicalRecordDetails.cs
namespace Nhom4_QLBA_API.Models
{
    public class MedicalRecordDetails
    {
        public int Id { get; set; }
        public int PatientId { get; set; }
        public string? Treatment { get; set; }
        public DateTime? CreatedAt { get; set; }
        public Patients? Patient { get; set; }
    }
}
